### ISSUE TITLE
Reframe docker compose installation

### ISSUE THREAD

**Issue by silentninja** [mathesar-foundation/mathesar#3166]
~~Based on our [installation plan](https://wiki.mathesar.org/en/projects/installation-improvements-plan-0_1_4), we won't be using supporting docker-compose installation. we should be deprecating it for now until we have a migration plan for the existing users who installed Mathesar using docker-compose~~

~~The following guides need to be marked as deprecated~~
- ~~[Docker compose installation](https://docs.mathesar.org/installation/docker-compose/)~~
- ~~["customizing docker compose" page](https://docs.mathesar.org/configuration/customize-docker-compose/)~~

### According to the new plan:
- Docker compose file and its self documenting nature should serve as a baseline for people looking at alternate but quite similar installation methods like k8s, docker swarm, etc. 
- Docs on docs.mathesar.org should be minimal and should refer to the docker compose file for details.

## Additional context
~~Blocked by #3072~~

**Comment by seancolsen** [comment]
Addressed in #3227
